packages/pds/src/services/repo/index.ts

```diff
@@ -159,6 +164,12 @@ export class RepoService {
     writes: PreparedWrite[],
     swapCommit?: CID,
   ): Promise<CommitData> {
+    // this is not in a txn, so this won't actually hold the lock,
+    // we just check if it is currently held by another txn
+    const available = await storage.lockAvailable()
+    if (!available) {
+      throw new ConcurrentWriteError()
+    }
```

Comment:
I take it this is defensive, i.e. not really required but allows us to fail early?

Reply:
yup exactly